express-delete/server.js

```diff
@@ -0,0 +1,45 @@
+import express from 'express';
+
+const app = express();
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.get('/api/grades', (req, res) => {
+  const newGrades = [];
+  for (const property in grades) {
+    newGrades.push(grades[property]);
+  }
+  res.json((newGrades));
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  const numberId = Number(req.params.id);
+  for (const property in grades) {
```

Comment:
It's not necessary to loop through the grades here. `delete grades[numberId]` is all that is needed since that will do nothing if the property doesn't exist.